napi_js_props.md

```diff
@@ -0,0 +1,225 @@
+# N-API - JavaScript Data Types and Values 
+
+N-API exposes a set of APIs to interrogate and set properties on JavaScript objects.
+Some of these types are documented under [Section 7](https://tc39.github.io/ecma262/#sec-operations-on-objects) 
+of the [ECMAScript Language Specification](https://tc39.github.io/ecma262/).
+
+Fundamentally, all properties in N-API can be represented in one of the following forms:
+- Named: a simple utf8 encoded string
+- Integer-Indexed: an index value represented by `uint32_t`
+- JavaScript Object: a complex JavaScript Object represented by `napi_value`
+
+N-API values are represented by the type `napi_value`.
+Any N-API API that requires a JavaScript object takes in a `napi_value`
+however, it's the caller's responsibility to make sure that the 
```

Comment:
Fixed- that was supposed to be multiple sentences (I had actually fixed it earlier but somehow that change got lost while I split up my docs into multiple branches 😞)